src/analysis/normed_space/lp_space.lean

```diff
@@ -202,6 +202,18 @@ begin
       exact real.rpow_le_rpow_of_exponent_ge' (norm_nonneg _) hi.le hq.le hpq' } }
 end
 
+lemma _root_.mem_ℓp_single [decidable_eq α] (p) (i : α) (a : E i) :
+  @mem_ℓp α E _ (λ j, if h : j = i then eq.rec a h.symm else 0) p :=
```

Comment:
This statement looks quite inreadable to me. Does it make sense to just inline this into the definition of `single`?